src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -223,6 +224,22 @@ private void CheckFileSystem()
             }
         }
 
+        private void ValidateLinuxSKUConfiguration()
+        {
+            var url = _environment.GetEnvironmentVariable(ScmRunFromPackage);
+
+            if (string.IsNullOrEmpty(url) && _environment.IsLinuxConsumptionOnAtlas() && _environment.IsManagedAppEnvironment())
+            {
+                _logger.LogWarning($"Zip deployment is not supported on a Linux Consumption app configured to use Managed Identity. Functions may not be indexed correctly.");
+
+                DiagnosticEventLoggerExtensions.LogDiagnosticEventInformation(
+                            _logger,
+                            DiagnosticEventConstants.UnsupportedZipDeploymentOnLinuxConsumptionwithMSIErrorCode,
+                            "Zip deployment is not supported on a Linux Consumption app configured to use Managed Identity.",
+                            DiagnosticEventConstants.UnsupportedZipDeploymentOnLinuxConsumptionwithMSIErrorCodeLink);
+            }
```

Comment:
@brettsam Do we want to stop the application here as we are doing in `CheckFileSystem()` (link below)? or is it sufficient to provide a warning via diagnostic event to the user? Wanted to double check so that we don't stop the application where it is not needed.

https://github.com/Azure/azure-functions-host/blob/88b610c61a6dc39e053f5fe7f902204e06aa1b41/src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs#L221

Reply:
You'd need to check that this is not on Managed App Environment (ACA), so the check would need to be `!_environment.IsManagedAppEnvironment()`